src/ConfigReader.php

```diff
@@ -0,0 +1,59 @@
+<?php
+
+namespace WMDE\Fundraising\Frontend;
+
+use FileFetcher\FileFetcher;
+use FileFetcher\FileFetchingException;
+use RuntimeException;
+
+/**
+ * @licence GNU GPL v2+
+ * @author Jeroen De Dauw < jeroendedauw@gmail.com >
+ */
+class ConfigReader {
+
+	private $fileFetcher;
+	private $baseConfigFilePath;
+	private $instanceConfigFilePath;
+
+	public function __construct( FileFetcher $fileFetcher, string $baseConfigFilePath, string $instanceConfigFilePath = null ) {
+		$this->fileFetcher = $fileFetcher;
+		$this->baseConfigFilePath = $baseConfigFilePath;
+		$this->instanceConfigFilePath = $instanceConfigFilePath;
+	}
+
+	/**
+	 * @return array
+	 * @throws RuntimeException
+	 */
+	public function getConfig(): array {
+		if ( $this->instanceConfigFilePath === null ) {
+			return $this->getFileConfig( $this->baseConfigFilePath );
+		}
+
+		return array_replace_recursive(
+			$this->getFileConfig( $this->baseConfigFilePath ),
+			$this->getFileConfig( $this->instanceConfigFilePath )
+		);
+	}
+
+	private function getFileConfig( string $filePath ): array {
+		$config = @json_decode( $this->getFileContents( $filePath ), true );
```

Comment:
Oh right... I was thinking about file_get_contents here...

Feel free to add a commit that removes this. I'll get to this during the evening if you have not already.